File: frontend/lib/models/activity.dart
class Activity {
  final int id;
  final String title;
  final String? description;
  final String? imageUrl;
  final String type;
  final String status;
  final DateTime startTime;
  final DateTime endTime;
  final double? discountRate;
  final double? minAmount;
  final double? maxDiscount;
  final int? participantLimit;
  final int participantCount;
  final int sortOrder;
  final DateTime createdAt;
  final DateTime updatedAt;

  Activity({
    required this.id,
    required this.title,
    this.description,
    this.imageUrl,
    required this.type,
    required this.status,
    required this.startTime,
    required this.endTime,
    this.discountRate,
    this.minAmount,
    this.maxDiscount,
    this.participantLimit,
    required this.participantCount,
    required this.sortOrder,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Activity.fromJson(Map<String, dynamic> json) {
    return Activity(
      id: json['id'] ?? 0,
      title: json['title'] ?? '',
      description: json['description'],
      imageUrl: json['imageUrl'],
      type: json['type'] ?? 'DISCOUNT',
      status: json['status'] ?? 'DRAFT',
      startTime: DateTime.parse(json['startTime'] ?? DateTime.now().toIso8601String()),
      endTime: DateTime.parse(json['endTime'] ?? DateTime.now().toIso8601String()),
      discountRate: json['discountRate']?.toDouble(),
      minAmount: json['minAmount']?.toDouble(),
      maxDiscount: json['maxDiscount']?.toDouble(),
      participantLimit: json['participantLimit'],
      participantCount: json['participantCount'] ?? 0,
      sortOrder: json['sortOrder'] ?? 0,
      createdAt: DateTime.parse(json['createdAt'] ?? DateTime.now().toIso8601String()),
      updatedAt: DateTime.parse(json['updatedAt'] ?? DateTime.now().toIso8601String()),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'imageUrl': imageUrl,
      'type': type,
      'status': status,
      'startTime': startTime.toIso8601String(),
      'endTime': endTime.toIso8601String(),
      'discountRate': discountRate,
      'minAmount': minAmount,
      'maxDiscount': maxDiscount,
      'participantLimit': participantLimit,
      'participantCount': participantCount,
      'sortOrder': sortOrder,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
    };
  }

  Activity copyWith({
    int? id,
    String? title,
    String? description,
    String? imageUrl,
    String? type,
    String? status,
    DateTime? startTime,
    DateTime? endTime,
    double? discountRate,
    double? minAmount,
    double? maxDiscount,
    int? participantLimit,
    int? participantCount,
    int? sortOrder,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Activity(
      id: id ?? this.id,
      title: title ?? this.title,
      description: description ?? this.description,
      imageUrl: imageUrl ?? this.imageUrl,
      type: type ?? this.type,
      status: status ?? this.status,
      startTime: startTime ?? this.startTime,
      endTime: endTime ?? this.endTime,
      discountRate: discountRate ?? this.discountRate,
      minAmount: minAmount ?? this.minAmount,
      maxDiscount: maxDiscount ?? this.maxDiscount,
      participantLimit: participantLimit ?? this.participantLimit,
      participantCount: participantCount ?? this.participantCount,
      sortOrder: sortOrder ?? this.sortOrder,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }

  bool get isActive => status == 'ACTIVE';
  bool get isExpired => DateTime.now().isAfter(endTime);
  bool get isFull => participantLimit != null && participantCount >= participantLimit!;
  bool get isOngoing => DateTime.now().isAfter(startTime) && DateTime.now().isBefore(endTime);

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;
    return other is Activity &&
        other.id == id &&
        other.title == title &&
        other.type == type &&
        other.status == status;
  }

  @override
  int get hashCode {
    return Object.hash(id, title, type, status);
  }

  @override
  String toString() {
    return 'Activity{id: $id, title: $title, type: $type, status: $status}';
  }
}

File: backend/src/main/java/com/playmate/entity/Activity.java
package com.playmate.entity;

import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "activities")
public class Activity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, length = 200)
    private String title;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @Column(name = "image_url", length = 500)
    private String imageUrl;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ActivityType type;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ActivityStatus status = ActivityStatus.DRAFT;
    
    @Column(name = "start_time", nullable = false)
    private LocalDateTime startTime;
    
    @Column(name = "end_time", nullable = false)
    private LocalDateTime endTime;
    
    @Column(name = "discount_rate", precision = 5, scale = 4)
    private BigDecimal discountRate;
    
    @Column(name = "min_amount", precision = 10, scale = 2)
    private BigDecimal minAmount;
    
    @Column(name = "max_discount", precision = 10, scale = 2)
    private BigDecimal maxDiscount;
    
    @Column(name = "participant_limit")
    private Integer participantLimit;
    
    @Column(name = "participant_count", nullable = false)
    private Integer participantCount = 0;
    
    @Column(name = "sort_order", nullable = false)
    private Integer sortOrder = 0;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    public enum ActivityType {
        DISCOUNT,    // 折扣活动
        BONUS,       // 奖励活动
        LIMITED_TIME // 限时活动
    }
    
    public enum ActivityStatus {
        DRAFT,       // 草稿
        ACTIVE,      // 进行中
        SUSPENDED,   // 暂停
        ENDED,       // 已结束
        CANCELLED    // 已取消
    }
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    
    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }
    
    public ActivityType getType() { return type; }
    public void setType(ActivityType type) { this.type = type; }
    
    public ActivityStatus getStatus() { return status; }
    public void setStatus(ActivityStatus status) { this.status = status; }
    
    public LocalDateTime getStartTime() { return startTime; }
    public void setStartTime(LocalDateTime startTime) { this.startTime = startTime; }
    
    public LocalDateTime getEndTime() { return endTime; }
    public void setEndTime(LocalDateTime endTime) { this.endTime = endTime; }
    
    public BigDecimal getDiscountRate() { return discountRate; }
    public void setDiscountRate(BigDecimal discountRate) { this.discountRate = discountRate; }
    
    public BigDecimal getMinAmount() { return minAmount; }
    public void setMinAmount(BigDecimal minAmount) { this.minAmount = minAmount; }
    
    public BigDecimal getMaxDiscount() { return maxDiscount; }
    public void setMaxDiscount(BigDecimal maxDiscount) { this.maxDiscount = maxDiscount; }
    
    public Integer getParticipantLimit() { return participantLimit; }
    public void setParticipantLimit(Integer participantLimit) { this.participantLimit = participantLimit; }
    
    public Integer getParticipantCount() { return participantCount; }
    public void setParticipantCount(Integer participantCount) { this.participantCount = participantCount; }
    
    public Integer getSortOrder() { return sortOrder; }
    public void setSortOrder(Integer sortOrder) { this.sortOrder = sortOrder; }
    
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
    
    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }
}